lib/jekyll/filters/url_filters.rb

```diff
@@ -29,6 +29,23 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces the path to the given asset relative to the assets directory at the
+      # domain root based on site.baseurl
+      #
+      # input - the asset file relative to the 'assets' directory. The string may also
+      #         include any subfolder(s) the asset may be a part of.
+      #           e.g. "main.css" or "css/main.css"
+      #
+      # Returns a path relative to the assets directory as a String.
+      def asset_url(input)
+        return if input.nil?
+        site = @context.registers[:site]
+        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        Addressable::URI.parse(
+          ensure_leading_slash(File.join(site.config["baseurl"], "assets", input.to_s))
```

Comment:
@pathawks can you elaborate while `File#join` is bad in this context?
@ashmaroli Depends. For systems work yes.  For URL's no, but that said, I think @pathawks is trying to say that Addressable has a method to deal with that kind of stuff, I know it has `#convert_path` but maybe there is something else I don't remember.